src/d3a/models/strategy/__init__.py

```diff
@@ -458,9 +458,9 @@ def event_market_cycle(self):
         self.event_responses = []
 
     def assert_if_trade_offer_price_is_too_low(self, market_id, trade):
-        if isinstance(trade.offer, Offer) and trade.offer.seller == self.owner.name:
-            offer = [o for o in self.offers.sold[market_id] if o.id == trade.offer.id][0]
-            assert trade.offer.energy_rate >= \
+        if trade.is_offer_trade and trade.offer_bid.seller == self.owner.name:
```

Comment:
Could you please add a type hint to the `trade` argument of this method? I suppose it is a `TradeBidOfferInfo` and not a `Trade` instance, right?